### PR TITLE
feat: add adaptSpectrum prop and handling to slider components

### PR DESCRIPTION
**Pull request by adamhari**
Adds an optional prop to support adjusting the slider spectrums based on the active brightness and saturation values.

https://user-images.githubusercontent.com/16565387/223205180-ec67340a-3952-46ab-9900-05d555538301.mp4

### CODE REVIEW

**Comment by alabsi91** [comment]
I should note that the `hueValue`, `saturationValue`, and `brightnessValue` are the color channels in the HSV/HSB color format. Hue is the same in both HSV and HSL formats, but the saturation and brightness values are different. Have you considered this in your PR? If not, I suggest creating a worklet function in a separate file that converts HSV to HSL color format. What are your thoughts?

**Comment by adamhari** [comment]
These changes account for the difference between HSV and HSL, but this is happening in the logic for each style object. I like the idea of extracting it to a separate reusable function.

Regarding your suggestion, I see that there is a method in `ColorKit` called `HSL_HSVA`. Do you know if we would be able to declare that as a worklet and use it as-is, or if we would need to duplicate it in order to keep the original in tact?

**Comment by alabsi91** [comment]
Maybe there is a way to reuse the function that I am not aware of. I know there is a `runOnUI` method, but I have never tried it.

**Comment by adamhari** [comment]
I went with a separate worklet function because the need to convert hsva --> hsla makes it an outlier from the rest of the `ColorKit` funcs and it seemed like a better option than introducing inconsistency or trying to make all of `ColorKit` work on the UI thread.

The brightness and opacity sliders work well, the hue slider is close to accurate but there is inconsistency because it needs to manage overlaying separate black and white overlays with transparency to give the effect of tweaking brightness and saturation. More work can be done there to improve the formula. The saturation slider has a similar issue because, we cannot represent saturation with a static gradient like we can with brightness and opacity.

**Comment by alabsi91** [comment]
At this point, I think we should consider changing the file name of `GlobalStyles` to something more suitable.

**Comment by adamhari** [comment]
How about we split it into `styles.ts`, `utils/helpers.ts`, and co-locate `CTX` with `ColorPicker.ts` (it doesn't really fit with either or need it's own file)

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+export function clamp(v: number, max: number) {
+  'worklet';
+  return Math.min(Math.max(v, 0), max);
+}
+
+export function hsva2Hsla(h: number, s: number, v: number, a = 1) {
+  'worklet';
+
```
Comment (alabsi91) [inline src/utils.ts:16]: - I noticed that sometimes you are using `brightnessValue.value / 100`, and other times you are using just `brightnessValue.value`.
- This function takes a range of `[0-100]` , and you will need to edit the inputs accordingly.
```ts
export function hsva2Hsla(h: number, s: number, v: number, a = 1) {
  'worklet';

  s = s / 100;
  v = v / 100;

  const l = ((2 - s) * v) / 2,
    sl = s * v,
    sln = l !== 0 && l !== 1 ? sl / (l < 0.5 ? l * 2 : 2 - l * 2) : sl;

  return `hsla(${h}, ${sln * 100}%, ${l * 100}%, ${a})`;
}
```

File: src/components/HueSlider.tsx [hunk 2]
```diff
@@ -113,7 +138,18 @@ export function HueSlider({ thumbShape, thumbSize, thumbColor, style = {}, verti
         style={[{ borderRadius }, style, thicknessStyle, { position: 'relative', borderWidth: 0, padding: 0 }]}
       >
         <Animated.Image source={require('../assets/Hue.png')} style={imageStyle} />
-        <Thumb {...{ channel: 'h', thumbShape, thumbSize: thumb_size, thumbColor: thumb_color, handleStyle, vertical }} />
+        <Animated.View style={[activeSaturationStyle, StyleSheet.absoluteFillObject]} />
```
Comment (alabsi91) [inline src/components/HueSlider.tsx:141]: - Both of the views you added do not match the parent's border radius. You just need to add `{ borderRadius }` to the style.

File: src/components/HueSlider.tsx [hunk 3]
```diff
@@ -52,6 +63,18 @@ export function HueSlider({ thumbShape, thumbSize, thumbColor, style = {}, verti
     };
   }, [thumbSize, vertical, reverse]);
 
+  const activeSaturationStyle = useAnimatedStyle(() => ({
+    backgroundColor: hsva2Hsla(
+      0,
+      0,
+      adaptSpectrum ? brightnessValue.value : 100,
+      adaptSpectrum ? 1 - saturationValue.value / 100 : 0
```
Comment (alabsi91) [inline src/components/HueSlider.tsx:71]: - For the sake of accuracy, I suggest adjusting the opacity manually. For example, you could subtract `0.2` from the calculated opacity on both the saturation and brightness layers.

**Comment by alabsi91** [comment]
- I was experimenting with the `hueSlider` to adjust the spectrum layers, and I found a way to achieve it in just one layer. Please give it a try and share your feedback with me.

```jsx
  const activeColorStyle = useAnimatedStyle(() => {
    const alpha1 = 1 - saturationValue.value / 100;
    const alpha2 = 1 - brightnessValue.value / 100;
    const opacity = Math.max(alpha2, alpha1);
    return { backgroundColor: hsva2Hsla(0, 0, brightnessValue.value, opacity) };
  });

// the layer component
{adaptSpectrum && <Animated.View style={[activeColorStyle, { borderRadius }, StyleSheet.absoluteFillObject]} />}

```

**Comment by alabsi91** [comment]
May we merge this PR now?

**Comment by adamhari** [comment]
Yeah, it's good to go

**Pull request merged**